pkg/mount/mount.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mount
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/errwrap"
+)
+
+// GetMountsForPrefix returns all mountpoints associated with a process mount namespace,
+// filtered by prefix. If no prefixPath is specified, all mounts are returned.
+// The special value 0 as pid argument is used to specify the current process.
+func GetMountsForPrefix(prefixPath string, pid uint) (Mounts, error) {
+	var procPath string
+	if pid == 0 {
+		procPath = "/proc/self/mountinfo"
+	} else {
+		procPath = fmt.Sprintf("/proc/%d/mountinfo", pid)
+	}
+
+	if prefixPath == "" {
+		prefixPath = "/"
+	}
+
+	mi, err := os.Open(procPath)
+	if err != nil {
+		return nil, err
+	}
+	defer mi.Close()
+
+	mnts, err := getMountsForPrefix(prefixPath, mi)
+	if err != nil {
+		return nil, errwrap.Wrap(fmt.Errorf("error getting sub-mounts under %q from mountinfo", prefixPath), err)
+	}
+
+	return mnts, nil
+}
+
+// getMountsForPrefix parses mi (/proc/<pid>/mountinfo) and returns mounts for path prefix
+func getMountsForPrefix(path string, mi io.Reader) (Mounts, error) {
+	var podMounts Mounts
+	sc := bufio.NewScanner(mi)
+	var (
+		mountID    int
+		parentID   int
+		mountPoint string
+		opt        map[string]struct{}
+	)
+
+	for sc.Scan() {
+		line := sc.Text()
+		lineResult := strings.Split(line, " ")
+		if len(lineResult) < 7 {
+			return nil, fmt.Errorf("Not enough fields from line %q: %+v", line, lineResult)
+		}
+
+		opt = map[string]struct{}{}
+		for i, s := range lineResult {
+			if s == "-" {
```

Comment:
Ack. Also linking to https://www.kernel.org/doc/Documentation/filesystems/proc.txt won't hurt :)